vulcanai2/models/cnn.py

```diff
@@ -105,7 +105,9 @@ def _create_network(self, **kwargs):
 
     def _create_classification_layer(self, dim, pred_activation):
         self.network_tail = DenseUnit(
-            dim, self.out_dim, activation=pred_activation)
+            in_features=dim,
+            out_features=self.out_dim,
+            activation=pred_activation)
 
```

Comment:
TODO: check for correct dimensions between input data and self.dimensions. Also remove self.in_dim from _create_network